src/InvestmentManager.sol

```diff
@@ -103,16 +105,17 @@ contract InvestmentManager is Auth, IInvestmentManager {
         require(state.pendingCancelDepositRequest != true, "InvestmentManager/cancellation-is-pending");
 
         state.pendingDepositRequest = state.pendingDepositRequest + _assets;
-        gateway.send(abi.encodePacked(uint8(MessagesLib.Call.EntryPoint), source));
         gateway.send(
-            abi.encodePacked(
-                uint8(MessagesLib.Call.IncreaseInvestOrder),
-                poolId,
-                vault_.trancheId(),
-                receiver,
-                poolManager.assetToId(asset),
-                _assets,
-                source
+            abi.encode(
+                source,
```

Comment:
I have decided  to  have something like 
```
  (
        address source; // metadata property
        .... // some other meadata property
       bytes message;      
  )
```

`source` is not part of the message. I don't think that this is needed  on the centrifuge side. This is something we care on our side. Also it allows us to extent this metadata structure without affecting the size of the message.

Reply:
Choose `encode` instead of `encodePacked` because I think it will be easier to decode instead of parsing byte by byte. We can use ` encodePacked` and in the `send` just get the first 20 bytes for the address and the remaining for the message.